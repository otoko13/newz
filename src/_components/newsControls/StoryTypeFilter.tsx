import React from 'react';
import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
import { EStoriesType } from '../newsService';

export interface IStoryTypeFilterProps {
    filter: string;
    onFilterChanged: (type: string) => void;
}

const StoryTypeFilter = (props: IStoryTypeFilterProps) => {

    function handleChange(_event: React.FormEvent<HTMLDivElement>, item?: IDropdownOption) {
        if (item) {
            props.onFilterChanged(item.key as string);
        }
    }

    return (
        <Dropdown
            className='StoryTypeFilter'
            selectedKey={props.filter}
            onChange={handleChange}
            options={[
                { key: EStoriesType.New.toString(), text: 'Newest' },
                { key: EStoriesType.Top.toString(), text: 'Top stories' },
                { key: EStoriesType.Best.toString(), text: 'Best stories' },
            ]}
            styles={{ dropdown: { width: 200 } }}
        />
    );
}

export default StoryTypeFilter;