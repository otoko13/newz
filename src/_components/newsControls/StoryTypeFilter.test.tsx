import { fireEvent, render } from '@testing-library/react';
import * as React from 'react';
import StoryTypeFilter, { IStoryTypeFilterProps } from './StoryTypeFilter';

let props: IStoryTypeFilterProps;

beforeEach(() => {
    props = {
        filter: 'new',
        onFilterChanged: jest.fn(),
    }
});

describe('snapshot', () => {
    it('should look sensible', () => {
        const { container } = render(<StoryTypeFilter {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});

describe('changing filter', () => {
    it('should call props.onFilterChanged with the correct new filter', () => {
        const { getByRole, getByTitle } = render(<StoryTypeFilter {...props} />);
        fireEvent.click(getByRole('listbox'));
        fireEvent.click(getByTitle('Best stories'));
        expect(props.onFilterChanged).toHaveBeenCalledWith('best');
    });
});