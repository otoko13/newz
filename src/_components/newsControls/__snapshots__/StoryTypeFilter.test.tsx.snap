// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should look sensible 1`] = `
<div
  class="ms-Dropdown-container StoryTypeFilter root-40"
>
  <div
    aria-expanded="false"
    aria-haspopup="listbox"
    class="ms-Dropdown dropdown-42"
    data-is-focusable="true"
    id="Dropdown0"
    role="listbox"
    tabindex="0"
  >
    <span
      aria-atomic="true"
      aria-invalid="false"
      aria-live="polite"
      aria-posinset="1"
      aria-selected="true"
      aria-setsize="3"
      class="ms-Dropdown-title title-43"
      id="Dropdown0-option"
      role="option"
    >
      Newest
    </span>
    <span
      class="ms-Dropdown-caretDownWrapper caretDownWrapper-44"
    >
      <i
        aria-hidden="true"
        class="ms-Dropdown-caretDown caretDown-58"
        data-icon-name="ChevronDown"
      />
    </span>
  </div>
</div>
`;
